src/sagemaker/model.py

```diff
@@ -448,21 +448,25 @@ def _upload_code(self, key_prefix, repack=False):
         local_code = utils.get_config_value('local.local_code', self.sagemaker_session.config)
         if self.sagemaker_session.local_mode and local_code:
             self.uploaded_code = None
-        else:
-            if not repack:
-                bucket = self.bucket or self.sagemaker_session.default_bucket()
-                self.uploaded_code = fw_utils.tar_and_upload_dir(session=self.sagemaker_session.boto_session,
-                                                                 bucket=bucket,
-                                                                 s3_key_prefix=key_prefix,
-                                                                 script=self.entry_point,
-                                                                 directory=self.source_dir,
-                                                                 dependencies=self.dependencies)
+        elif not repack:
+            bucket = self.bucket or self.sagemaker_session.default_bucket()
+            self.uploaded_code = fw_utils.tar_and_upload_dir(session=self.sagemaker_session.boto_session,
+                                                             bucket=bucket,
+                                                             s3_key_prefix=key_prefix,
+                                                             script=self.entry_point,
+                                                             directory=self.source_dir,
+                                                             dependencies=self.dependencies)
 
         if repack:
-            self.repacked_model_data = utils.repack_model(inference_script=self.entry_point,
-                                                          source_directory=self.source_dir,
-                                                          model_uri=self.model_data,
-                                                          sagemaker_session=self.sagemaker_session)
+            bucket = self.bucket or self.sagemaker_session.default_bucket()
+            self.repacked_model_data = 's3://' + os.path.join(bucket, key_prefix, 'model.tar.gz')
```

Comment:
Should this assignment happen after the repack_model function finishes successfully?

This can be a problem in a notebook, where state is stored even with exceptions happening.